tutorials/persistance-comparison-series/Sqlite/sqlite.md

```diff
@@ -0,0 +1,390 @@
+
+Whenever the game starts (2), we want to read the current hit count from the persistence and initialize `hitCount` accordingly (3). This is done in the `Start()` method that is called whenever a scene is loaded for each game object this script is attached to.
+
+The second part to this is saving changes, which we want to do whenever we register a mouse click. The Unity message for this is `OnMouseDown()` (4). This method gets called every time the `GameObject` that this script is attached to is clicked (with a left mouse click). In this case, we increment the `hitCount` (5) which will eventually be saved by the various options shown in this tutorials series.
+
+## Sqlite
+
+(See `SqliteExampleSimple.cs` in the repository for the finished version.)
+
+Now let's make sure our hit count gets persisted so we can continue playing the next time we start the game.
+
+Sqlite is not included per default in a new Unity project and also not available directly via the Unity package manager. We have to install two components to start using it.
+
+First, head over to [https://sqlite.org/download.html](https://sqlite.org/download.html) and choose the `Precompiled Binaries` for your operating system. Unzip it and add the two files - `sqlite3.def` and `sqlite3.dll` - to the `Plugin` folder in your Unity project.
+
+Then, open a file explorer in your Unity Hub installation directory, head to the following sub directory:
+
+```
+Unity/Hub/Editor/2021.2.11f1/Editor/Data/MonoBleedingEdge/lib/mono/unity
+```
+
+In there you will find the file `Mono.Data.Sqlite.dll` which also needs to be moved to the `Plugins` folder in your Unity project. The result when going back to the Editor should look like this:
+
+<img src="images/05_sqlite_in_project_hierarchy.png" alt="Capsule in scene"/>
+
+Now that the preparations are finished we want to add our first script to the capsule. Similar to the `HitCountExample.cs` create a new `C# script` and name it `SqliteExampleSimple`.
+
+When opening it, the first thing we want to do, is importing Sqlite by adding `using Mono.Data.Sqlite;` and `using System.Data;` at the top of the file (1).
+
+Next we will look at how to save whenever the hit count is changed, which happens during `OnMouseDown()`. Similar to the `FileStream` and the `BinaryWriter` we have seen in the previous parts of this series we need some way to write the data to where it should go. In case of Sqlite that means we need a connection to the database. This is offered by the Sqlite library via the [`IDbConnection`](https://docs.microsoft.com/en-us/dotnet/api/system.data.idbconnection?view=net-6.0) class (2) which represents an open connection to the database. Since we will need a connection for loading the data later on again, we will extract opening a database connection into into another function and call it `private IDbConnection CreateAndOpenDatabase()` (3).
```

Comment:
```suggestion
Next we will look at how to save whenever the hit count is changed, which happens during `OnMouseDown()`. Similar to the `FileStream` and the `BinaryWriter` we have seen in the previous parts of this series we need some way to write the data to where it should go. In case of Sqlite that means we need a connection to the database. This is offered by the Sqlite library via the [`IDbConnection`](https://docs.microsoft.com/en-us/dotnet/api/system.data.idbconnection?view=net-6.0) class (2) which represents an open connection to the database. Since we will need a connection for loading the data later on again, we will extract opening a database connection into another function and call it `private IDbConnection CreateAndOpenDatabase()` (3).
```